tests/integration-tests/tests/health_check/test_scheduled_events.py

```diff
@@ -0,0 +1,223 @@
+    cluster_config = pcluster_config_reader(initial_queue_size=num_compute_nodes)
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    events_substack_path = os.path.join("..", "..", "cloudformation", "scheduled-events-substack.cfn.json")
+    # Test that the default rule is able to match a sample AWS health event
+    _test_real_event_pattern(region, events_substack_path, test_datadir)
+
+    # We cannot push real scheduled events to event bus
+    # Create a fake rule that will be trigger by a fake health event in order to test end-to-end behavior of the cluster
+    queue_info = _get_health_queue_info(
+        region, cluster.cfn_resources.get("ScheduledEventsSubstack"), remote_command_executor
+    )
+    stack_name = "fake-rule"
+    path = str(test_datadir / "fake-event-rule.cfn.json")
+    scheduled_event_substack_generator.generate_scheduled_event_rule(
+        path, make_real_rule=False, input_parameters=queue_info
+    )
+    stack_factory(region, stack_name, path, [])
+
+    _test_health_event_behavior(
+        cluster.cfn_name,
+        region,
+        scheduler,
+        remote_command_executor,
+        scheduler_commands,
+        num_compute_nodes,
+        health_check_enabled=True,
+    )
+    remote_command_executor.run_remote_script(str(test_datadir / "disable_health_check.sh"))
+    _test_health_event_behavior(
+        cluster.cfn_name,
+        region,
+        scheduler,
+        remote_command_executor,
+        scheduler_commands,
+        num_compute_nodes,
+        health_check_enabled=False,
+    )
+
+    assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher"])
+
+
+def _test_real_template(region, events_substack_path, test_datadir):
+    _test_real_event_pattern(region, events_substack_path, test_datadir)
+    _assert_real_template_syntax(events_substack_path)
+
+
+def _test_real_event_pattern(region, events_substack_path, test_datadir):
+    logging.info("Testing that ScheduledEventRule will match a real AWS scheduled event notification.")
+    with open(events_substack_path) as substack_file:
+        events_substack = json.load(substack_file)
+    real_event_pattern = (
+        events_substack.get("Resources").get("ScheduledEventRule").get("Properties").get("EventPattern")
+    )
+    with open(str(test_datadir / "sample_scheduled_event.json")) as sample_event_file:
+        sample_event = json.load(sample_event_file)
+    events_client = boto3.client("events", region_name=region)
+    response = events_client.test_event_pattern(
+        EventPattern=json.dumps(real_event_pattern), Event=json.dumps(sample_event)
+    )
+    assert_that(response.get("Result")).is_true()
+    logging.info("ScheduledEventRule successfully matched a real AWS scheduled event notification.")
+
+
+def _assert_real_template_syntax(events_substack_path):
```

Comment:
I'm ok with leaving this here for now but we could have coded this as a unit test for cloudformation running in travis/tox